js/coinsbit.js

```diff
@@ -0,0 +1,742 @@
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        request['market'] = market['id'];
+        let ticker = {};
+        if (market['spot']) {
+            const response = await this.spotPublicGetTicker (request);
+            //     {
+            //          success: true,
+            //          message: "",
+            //          result: {
+            //            bid: '46504.7412435',
+            //            ask: '46511.54203562',
+            //            open: '47030.33',
+            //            high: '47549.99',
+            //            low: '45571.08',
+            //            last: '46459.99',
+            //            volume: '8372.29022355',
+            //            deal: '388090406.73607745',
+            //          },
+            //          code: "200"
+            //     }
+            ticker = this.safeValue (response, 'result', {});
+        }
+        return this.parseTicker (ticker, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        const [defaultType, request] = this.handleMarketTypeAndParams ('fetchTickers', undefined, params);
+        await this.loadMarkets ();
+        let tickers = {};
+        if (defaultType === 'spot') {
+            const response = await this.spotPublicGetTickers (request);
+            //     {
+            //          success: true,
+            //          message: "",
+            //          result: {
+            //            BTC_USDT: {
+            //               at: '1640028811',
+            //               ticker: {
+            //                 bid: '46448.49682539',
+            //                 ask: '46458.41093569',
+            //                 open: '46687.39',
+            //                 high: '47550',
+            //                 low: '45584.27',
+            //                 last: '46462.77',
+            //                 vol: '32.56275846',
+            //                 deal: '1510574.62551143',
+            //                 change: '-0'
+            //               }
+            //            },
+            //            ...
+            //          },
+            //          code: "200"
+            //     }
+            tickers = this.safeValue (response, 'result', {});
+        }
+        return this.parseTickers (tickers, undefined, params);
+    }
+
+    // This modified parseTickers method needs to exist in base, to keep the KYES & pass them through parseTickers
+    parseTickers (tickers, symbols = undefined, params = {}) {
+        const result = [];
+        const values = Object.values (tickers || []);
+        const keys = Object.keys (tickers);
+        for (let i = 0; i < values.length; i++) {
+            result.push (this.extend (this.parseTicker (values[i], undefined, keys[i]), params));
+        }
+        return this.filterByArray (result, 'symbol', symbols);
+    }
+
+    parseTicker (ticker, market = undefined, key = undefined) {
```

Comment:
We should not change the signature of `parseTicker`.

Reply:
Well, how this should be solved so? I can't find a solution if so

Reply:
In all such cases we should resolve the symbol and the market from the symbol outside of `parseTicker` and parse the resolved market as the second argument to `parseTicker`. We will later unify this in `parseTickers` in the base class.

Like this: 

- https://github.com/ccxt/ccxt/blob/master/js/coinbasepro.js#L539
- https://github.com/ccxt/ccxt/blob/master/js/hitbtc3.js#L618
- https://github.com/ccxt/ccxt/blob/master/js/luno.js#L410

Reply:
I see now - so you mean directly parsing ticker inside `fetchTickers` (without sending it to `parseTickers`), right?

Reply:
Yes, we will later detach those loops from fetchTickers.

Reply:
done